blocks/carousel/carousel.js

```diff
@@ -1,103 +1,130 @@
+  swiperPagination.classList.add('swiper-pagination', 'slideshow__pagination', 'swiper-pagination-clickable', 'swiper-pagination-bullets');
+
+  props.items.forEach((ele, index) => {
+    const span = document.createElement('span');
+    span.classList.add('slideshow__pagination-bullet');
+
+    if (index === 0) {
+      span.classList.add('slideshow__pagination-bullet--active');
     }
-    return '';
-  })()}
-
-                                    <span class="lazy-image__caption">
-                                      <p>${item.label}</p>
-                                    </span>
-                                  </figcaption>
-                                </figure>
-                              </div>
-                            </div>
-                          </div>
-                        </div>
-
-                        <div class="thumbnail-slideshow-item__media-indicator"></div>
-                      </div>
-                    </li>
-
-                    `).join('')}
-
-                  </ul>
-
-                  <div class="swiper-button-next slideshow__navigation-button slideshow__navigation-button--next" role="navigation">
-                    <span class="icon icon-chevron-right"></span>
-                  </div>
-
-                  <div class="swiper-button-prev slideshow__navigation-button slideshow__navigation-button--right swiper-button-disabled" role="navigation">
-                    <span class="icon icon-chevron-left"></span>
-                  </div>
-
-                  <div class="slideshow__pagination-wrapper">
-                    <div class="slideshow__pagination-bar">
-                      <div class="swiper-pagination slideshow__pagination swiper-pagination-clickable swiper-pagination-bullets" role="navigation">
-                        ${props.items.map((_, index) => `
-                          <span class="slideshow__pagination-bullet ${index === 0 ? 'slideshow__pagination-bullet--active' : ''}"></span>
-                        `).join('')}
-                      </div>
-
-                      <span class="slideshow__pagination-bullet slideshow__pagination-bullet--animated" style="width: ${100 / props.items.length}%; left: 0%;"></span>
-                    </div>
-                  </div>
-
-                </div>
-              </div>
-            </div>
-          </div>
-        </div>
-      </div>
-    </div>
-  `;
+
+    swiperPagination.appendChild(span);
+  });
+
+  slideShowBar.appendChild(swiperPagination);
+
+  const paginationBullet = document.createElement('span');
+  paginationBullet.classList.add('slideshow__pagination-bullet', 'slideshow__pagination-bullet--animated');
+  paginationBullet.style.width = `${(100 / props.items.length)}%`;
+
+  slideShowBar.appendChild(paginationBullet);
+  slideShowWrapper.appendChild(slideShowBar);
+
+  carouselDiv.appendChild(slideShowWrapper);
+}
+
+function template(props) {
+  const carouselDiv = document.createElement('div');
```

Comment:
Not sure if we need this additional nesting. Are there additional styles needed here or can the carousel block itself become the grid container?

Reply:
I think yes, will check and update accordingly